src/nft-card-front.ts

```diff
@@ -133,49 +154,50 @@ export class NftCardFrontTemplate extends LitElement {
       }
     `
   }
+  @property({type: Object}) public asset!: OpenSeaAsset
+  @property({type: Boolean}) public isOwnedByAccount!: boolean
+  @property({type: String}) public account!: string
+  @property({type: Boolean}) public horizontal!: boolean
+  @property({type: Object}) public state!: State
 
-  /**
+  @property({type: Boolean}) private isForSale: boolean = false
+  @property({type: Boolean}) private isLoading: boolean = true
+  @property({type: Object}) private lastSaleData?: LastSaleData
+
+  /*
    * EventHandler - Dispatch event allowing parent to handle click event
-   *
-   * @param e the event context
-   * @param type the event context
-   * @param data the event context
+   * TODO: Add EventType
    */
-  public eventHandler(e, type, data = {}) {
-    const event = new CustomEvent('new-event', {
+  public eventHandler(_event: any, type: string, data = {}) {
+    const newEvent = new CustomEvent('new-event', {
       detail: {
         type,
         data
       }
     })
-    this.dispatchEvent(event)
-  }
-
-  public connectedCallback() {
-    super.connectedCallback()
-    // setTimeout(() => this.eventHandler('','flip'), 10);
-    // console.warn('I flip card for testing remove me later')
+    this.dispatchEvent(newEvent)
   }
 
-  public updated(changedProperties: array) {
+  public updated(changedProperties: Map<string, string>) {
     // Assumption: If the traitData gets updated we should rebuild the
     // traits object that populates UI
     // Assumption: This will ONLY get called once per refresh
-    changedProperties.forEach((oldValue, propName) => {
+    changedProperties.forEach((_oldValue: string, propName: string) => {
       if (propName === 'asset') {
         // We got the data so we are done loading
-        this.loading = false
+        this.isLoading = false
 
         // Check for a sell order to populate the UI with the sell information
-        if (this.asset.sellOrders.length > 0) {
-          this.canBuy = true
-          const order = this.asset.sellOrders[0]
-          const token = order.paymentTokenContract
-          const currentPrice = order.currentPrice.toFixed() / Math.pow(10, token.decimals)
-          const expires = order.expirationTime.toFixed()
+        // TODO: We will be using lastSale here once added to SDK
+        if (this.asset.sellOrders!.length > 0) {
```

Comment:
i think you meant `if (this.asset.sellOrders?.length`. You need typescript 3.8 or higher to use the question marks